src/main/cpp/src/parse_uri.cu

```diff
@@ -600,8 +602,15 @@ uri_parts __device__ validate_uri(const char* str,
     slash -= skip;
   }
 
-  // no more string to parse is an error
+  // no more string to parse is generally an error ...
   if (len <= 0) {
+    // unless we had no scheme
+    if (!has_scheme) {
+      // Entirely empty or we only had a fragment, which is a valid URI reference.
+      // This is equivalent to having a path that is present but empty, so mark it ok too
+      ret.valid |= (1 << static_cast<int>(URI_chunks::PATH));
+      return ret;
```

Comment:
This will not set `ret.valid` for anything but path, but before it would be invalid for other chunk types. Should we zero valid before setting this bit?

Reply:
Well, we could have had a fragment.  E.g. the URI "#bob" satisfies this condition.  However in this case the fragment is NULL for some reason, so yes, we should clear the valid flag first.